contracts/test/behaviour/strategy.js

```diff
@@ -0,0 +1,336 @@
+const { expect } = require("chai");
+const { Wallet } = require("ethers");
+
+const { units, usdcUnits } = require("../helpers");
+const { impersonateAndFund } = require("../../utils/signers");
+const { parseUnits } = require("ethers/lib/utils");
+
+/**
+ *
+ * @param {*} context a function that returns a fixture with the additional properties:
+ * - strategy: the strategy to test
+ * - assets: array of tokens to test
+ * - vault: Vault or OETHVault contract
+ * - harvester: Harvester or OETHHarvester contract
+ * @example
+    shouldBehaveLikeStrategy(() => ({
+      ...fixture,
+      strategy: fixture.convexFrxEthWethStrategy,
+      assets: [fixture.weth, fixture.frxETH],
+      vault: fixture.oethVault,
+      harvester: fixture.oethHarvester,
+    }));
+ */
+const shouldBehaveLikeStrategy = (context) => {
+  describe("Strategy behaviour", () => {
+    it("Should have vault configured", async () => {
+      const { strategy, vault } = await context();
+      expect(await strategy.vaultAddress()).to.equal(vault.address);
+    });
+    it("Should be a supported asset", async () => {
+      const { assets, strategy } = await context();
+
+      for (const asset of assets) {
+        expect(await strategy.supportsAsset(asset.address)).to.be.true;
```

Comment:
nit: Could probably add check for random assets (other than ones defined in `assets`) to make sure `strategy.supportAsset()` returns for those

Reply:
do you mean a negative test like this?
```js
const unsupportedAssets = [crv, cvx];
for (const asset of unsupportedAssets) {
  expect(await strategy.supportsAsset(asset.address)).to.be.false;
}
```

the other option is we use a more comprehensive list of assets and then filter out the supported ones

Reply:
FraxETH strategy for instance doesn't support stETH or rETH. Those kind of things would be hard to cover if we use hardcoded value of assets though.

Probably just have something like:
```
const randomAssets = [usdt, usdc, dai, weth, reth, steth, frxeth, cvx, crv];
for (const asset of randomAssets) {
  if (asset in assets) { continue; } // Since `assets` already has a list of supported assets
  expect(await strategy.supportsAsset(asset.address)).to.be.false;
}
```
Would be better to wrap it with a `Promise.all` as well

Reply:
I've gone with a separate test for the negative case
```js
it("Should NOT be a supported asset", async () => {
    const {
      assets,
      strategy,
      usdt,
      usdc,
      dai,
      weth,
      reth,
      stETH,
      frxETH,
      crv,
      cvx,
    } = await context();

    const randomAssets = [
      usdt,
      usdc,
      dai,
      weth,
      reth,
      stETH,
      frxETH,
      cvx,
      crv,
    ];
    for (const asset of randomAssets) {
      if (assets.includes(asset)) {
        continue; // Since `assets` already has a list of supported assets
      }
      expect(await strategy.supportsAsset(asset.address)).to.be.false;
    }
  });
  ```

Reply:
I've pushed the above change but it ended up being a bigger refactor of the strategy behaviour in order to handle the frxETH strategy's weth asset which can deposit but not checkBalance or withdraw